modules/integration/tests-ui/src/test/java/org/wso2/carbon/appmanager/integration/ui/Util/APPMPublisherRestClient.java

```diff
@@ -337,7 +336,46 @@ private HttpResponse addRole(String roles, String appId) throws Exception {
 		}
 	}
 
-    /**
+	/**
+	 * this method add the policy group for the app and return the policy group id
+	 * @param policyPartial
+	 * @param anonymousAccessToUrlPattern
+	 * @param policyGroupName
+	 * @param throttlingTier
+	 * @param objPartialMappings
+	 * @param policyGroupDesc
+	 * @return policyGroupId
+	 * @throws Exception
+	 */
+	public String savePolicyGroup(String policyPartial, String anonymousAccessToUrlPattern, String policyGroupName,
+								  String throttlingTier, String objPartialMappings, String policyGroupDesc) throws Exception {
+
+		String payLoad = "policyPartial="+ URLEncoder.encode(policyPartial,"UTF-8") +
+				"anonymousAccessToUrlPattern=" + anonymousAccessToUrlPattern +
+				"&policyGroupName=" + policyGroupName +
+				"&throttlingTier=" + throttlingTier +
+				"&objPartialMappings=" + objPartialMappings +
+				"&policyGroupDesc=" + policyGroupDesc +
+				"&userRoles";
+
+		this.requestHeaders.put("Content-Type", "application/x-www-form-urlencoded");
+
+		HttpResponse response =
+				HttpRequestUtil.doPost(new URL(backEndUrl +
+								"/publisher/api/entitlement/policy/partial/policyGroup/save"), payLoad,
+						requestHeaders);
+		if (response.getResponseCode() == 200) {
+			JSONObject saveObject = new JSONObject(response.getData());
+			String policyGroupId = saveObject.optJSONObject("response").getString("id");
+			return policyGroupId;
+		} else {
```

Comment:
This code segment has been refactored and this code line is not in here any more.